### PR TITLE
Operating info fix

### PR DESCRIPTION
**Pull request by daniel-sousa-me**
Third time's a charm (I hope...)

### CODE REVIEW

**Comment by tobymao** [comment]
@roseundy please review

File: lib/engine/step/g_1860/dividend.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ def process_dividend(action)
 
           entity.operating_history[[@game.turn, @round.round_num]] = OperatingInfo.new(
             routes,
-            action,
+            (@game.insolvent?(entity) ? @game.actions.last : action),
```
Comment (roseundy) [inline lib/engine/step/g_1860/dividend.rb:30]: Why are you doing this? An insolvent company has a withhold dividend action. Why is this needed?

File: lib/engine/game/g_1860.rb [hunk 2]
```diff
@@ -388,7 +388,7 @@ def check_nationalize?
       end
 
       def get_or_revenue(info)
-        info.dividend.kind == 'withhold' ? 0 : info.revenue
+        !info.dividend.is_a?(Action::Dividend) || info.dividend.kind == 'withhold' ? 0 : info.revenue
```
Comment (roseundy) [inline lib/engine/game/g_1860.rb:391]: What problem is this solving? Why wouldn't a dividend not have a dividend action?

File: lib/engine/step/g_1860/dividend.rb (hunk 1 above)
Comment (daniel-sousa-me) [reply lib/engine/step/g_1860/dividend.rb:30]: It has a phantom action that doesn't have an id

File: lib/engine/game/g_1860.rb (hunk 2 above)
Comment (daniel-sousa-me) [reply lib/engine/game/g_1860.rb:391]: When the Dividend step is skipped (ie, no train) the OperatingInfo gets the previous action so that the links on the spreadsheet work (see the mentioned issue)

File: lib/engine/step/g_1860/dividend.rb (hunk 1 above)
Comment (crericha) [reply lib/engine/step/g_1860/dividend.rb:30]: Maybe a dumb question, but why can't 1860 use the same logic as `Dividend::skip!`? Why is the phantom action needed here but not there?

Comment (roseundy) [reply lib/engine/step/g_1860/dividend.rb:30]: Because unlike normal dividend skip operations, there is a route being run for an insolvent company, but the corp is forced to withhold (and the amount isn't zero).

File: lib/engine/game/g_1860.rb (hunk 2 above)
Comment (roseundy) [reply lib/engine/game/g_1860.rb:391]: But the amount isn't 0 in this case. It's the revenue the leased train runs for.

File: lib/engine/step/g_1860/dividend.rb (hunk 1 above)
Comment (roseundy) [reply lib/engine/step/g_1860/dividend.rb:30]: Who is using the id on the actions stored in operating_history?

File: lib/engine/game/g_1860.rb (hunk 2 above)
Comment (tobymao) [reply lib/engine/game/g_1860.rb:391]: you can jump to the action in the spreadsheet

Comment (daniel-sousa-me) [reply lib/engine/game/g_1860.rb:391]: How? Where is that action stored? It just looks like a dangling object

File: lib/engine/step/g_1860/dividend.rb (hunk 1 above)
Comment (tobymao) [reply lib/engine/step/g_1860/dividend.rb:30]: meant to put this here, but you can jump to an action in the spreadsheet which is why it needs an id

Comment (roseundy) [reply lib/engine/step/g_1860/dividend.rb:30]: I think I'm OK with these changes then.

File: lib/engine/game/g_1860.rb (hunk 2 above)
Comment (tobymao) [reply lib/engine/game/g_1860.rb:391]: @roseundy @daniel-sousa-me can you guys clarifiy this particular line? why is the dividend not an action::dividend, what type is it?

Comment (roseundy) [reply lib/engine/game/g_1860.rb:391]: It's whatever @game.actions.last is on line 30 of step/1860/dividend.rb

Comment (daniel-sousa-me) [reply lib/engine/game/g_1860.rb:391]: Yeah, it's the action that triggered a skip on the Dividends

Comment (daniel-sousa-me) [reply lib/engine/game/g_1860.rb:391]: This is not specific to 1860 at all. It is also changed in `step/dividend.rb`. 1860 just was problematic because I didn't notice it was completely bypassing the parent code

Comment (roseundy) [reply lib/engine/game/g_1860.rb:391]: Hmm, pretty sure I borrowed that code from a different game.

File: lib/engine/step/g_1860/dividend.rb (hunk 1 above)
Comment (roseundy) [reply lib/engine/step/g_1860/dividend.rb:30]: What revenue gets reported on the spreadsheet?

**Review by roseundy — APPROVED** [review]
